tools/sc3/main.go

```diff
@@ -0,0 +1,201 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/GeoNet/delta/internal/metadb"
+	"github.com/GeoNet/delta/resp"
+)
+
+func main() {
+
+	var base string
+	flag.StringVar(&base, "base", "../..", "delta base files")
+
+	var output string
+	flag.StringVar(&output, "output", "keys", "output sc3 configuration directory")
+
+	var networks string
+	//TODO: this could be managed via configuration elsewhere.
+	flag.StringVar(&networks, "networks", "AK,CB,CH,EC,FI,HB,KI,MN,NX,NZ,OT,RT,SC,SI,SM,SP,SX,TP,TR,WL", "comma separated list of networks to use")
+
+	var exclude string
+	flag.StringVar(&exclude, "exclude", "", "comma separated list of stations to skip (either NN_SSSS, or simply SSSS)")
+
+	var include string
+	flag.StringVar(&include, "include", "", "comma separated list of external stations to add (requires NN_SSSS)")
+
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "\n")
+		fmt.Fprintf(os.Stderr, "Build a set of sc3 key files\n")
+		fmt.Fprintf(os.Stderr, "\n")
+		fmt.Fprintf(os.Stderr, "Usage:\n")
+		fmt.Fprintf(os.Stderr, "\n")
+		fmt.Fprintf(os.Stderr, "  %s [options]\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "\n")
+		fmt.Fprintf(os.Stderr, "Options:\n")
+		fmt.Fprintf(os.Stderr, "\n")
+		flag.PrintDefaults()
+		fmt.Fprintf(os.Stderr, "\n")
+	}
+
+	flag.Parse()
+
+	check := make(map[string]interface{})
+	for _, n := range strings.Split(networks, ",") {
+		check[strings.TrimSpace(strings.ToUpper(n))] = true
+	}
+
+	var unmatch []string
+	for _, p := range strings.Split(exclude, ",") {
+		if s := func() string {
+			switch r := strings.Split(strings.ToUpper(p), "_"); len(r) {
+			case 2:
+				return strings.TrimSpace(r[0]) + "_" + strings.TrimSpace(r[1])
+			case 1:
+				if strings.ContainsAny(r[0], "*?\\[") {
+					return strings.TrimSpace(r[0])
+				}
+				return "*_" + strings.TrimSpace(r[0])
+			default:
+				return ""
+			}
+		}(); s != "" {
+			unmatch = append(unmatch, s)
+		}
+	}
+
+	now := time.Now().UTC()
+
+	globals := make(map[string]Global)
+	autoPicks := make(map[string]AutoPick)
+	stations := make(map[string]Station)
+
+	db := metadb.NewMetaDB(base)
+
+	dbStations, err := db.Stations()
+	if err != nil {
+		log.Fatalf("unable to load metadata stations: %v", err)
+	}
+
+	for _, station := range dbStations {
+		if _, ok := check[station.Network]; !ok {
+			continue
+		}
+
+		installations, err := db.Installations(station.Code)
+		if err != nil {
+			log.Fatalf("unable to load metadata installations %s: %v", station.Code, err)
+		}
+
+		network, err := db.Network(station.Network)
+		if network == nil || err != nil {
```

Comment:
I think this makes it a little harder to read, not a deal breaker.

```go
if err != nil {
   log.Fatalf(...)
}
if network == nil {
   continue
}
```

See a few below as well (and the one on L186 is pretty ugly).